engine/api/migrate/default_tags.go

```diff
@@ -0,0 +1,89 @@
+package migrate
+
+import (
+	"database/sql"
+
+	"github.com/go-gorp/gorp"
+
+	"github.com/ovh/cds/engine/api/cache"
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/engine/api/project"
+	"github.com/ovh/cds/engine/api/workflow"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+// DefaultTagsMigration useful to set default tags to git.branch git.author
+func DefaultTagsMigration(store cache.Store, DBFunc func() *gorp.DbMap, u *sdk.User) {
+	defaultTagsKey := "default_tags"
+	db := DBFunc()
+
+	log.Info("DefaultTagsMigration> Begin")
+
+	projs, errP := project.LoadAll(db, store, u)
+	if errP != nil {
+		log.Warning("DefaultTagsMigration> Cannot load all project: %s", errP)
+		return
+	}
+
+	for _, proj := range projs {
+		workflows, err := workflow.LoadAll(db, proj.Key)
+		if err != nil {
+			log.Warning("DefaultTagsMigration> Cannot load all workflows for project %s : %s", proj.Key, err)
+			continue
+		}
+
+		for _, wf := range workflows {
+			tx, errTx := db.Begin()
+			if errTx != nil {
```

Comment:
don't you have to lock the workflow ?

Reply:
I only lock the metadata of the workflow

Reply:
fine